src/api/features/features.api.js

```diff
@@ -49,83 +86,325 @@ export function extractOlFeatureGeodesicCoordinates(feature) {
+ * @param {Number} config.featureCount How many feature should be returned (maximum)
+ * @returns {Promise<LayerFeature[]>}
+ * @throws GetFeatureInfoError If any part of the input is not valid
+ */
+async function identifyOnExternalLayer(config) {
+    const {
+        layer = null,
+        coordinate = null,
+        resolution = null,
+        tolerance = null,
+        projection = null,
+        lang = null,
+        featureCount = null,
+    } = config
+    if (!layer) {
+        throw new GetFeatureInfoError('Missing layer')
+    }
+    if (!layer.hasTooltip || !layer.getFeatureInfoCapability) {
+        throw new GetFeatureInfoError(`Layer ${layer.id} can't be getFeatureInfo requested`)
+    }
+    if (coordinate === null) {
+        throw new GetFeatureInfoError('Coordinate are required to perform a getFeatureInfo request')
+    }
+    if (resolution === null) {
+        throw new GetFeatureInfoError(
+            'Map resolution is required to perform a getFeatureInfo request'
+        )
+    }
+    if (featureCount === null) {
+        throw new GetFeatureInfoError(
+            'A feature count is required to perform a getFeatureInfo request'
+        )
+    }
+    if (tolerance === null) {
+        throw new GetFeatureInfoError('A tolerance is required to perform a getFeatureInfo request')
+    }
+    if (lang === null) {
+        throw new GetFeatureInfoError('A lang is required to build a getFeatureInfo request')
+    }
+    if (
+        !projection ||
+        !layer.availableProjections.some(
+            (availableProjection) => availableProjection.epsg === projection.epsg
+        )
+    ) {
+        throw new GetFeatureInfoError(
+            `Wanted projection ${projection?.epsg} isn't supported by layer ${layer.externalLayerId}, possible projections are ${layer.availableProjections.map((projection) => projection.epsg).join(', ')}`
+        )
+    }
+    if (layer instanceof ExternalWMSLayer) {
+        // A "fake" 100x100 pixel extent will be calculated around the given coordinate to build the request
+        // (with the request asking for the pixel 50:50). This is done because, after many attempts to give
+        // the current map viewport and click pixel position without any positive results, I looked at what
+        // OpenLayers does under the hood, and that's exactly their approach (and the approach mf-geoadmin3
+        // had, because it relied on the Ol class to do exactly that). And as we wanted to be as
+        // framework-agnostic as possible, I couldn't get myself to pass the OL instance of the WMS source
+        // to this function to use the utils directly, and preferred to implement it on my own.
+        const requestExtent = createPixelExtentAround({
+            size: GET_FEATURE_INFO_FAKE_VIEWPORT_SIZE,
+            coordinate,
+            projection,
+            resolution,
+            rounded: true,
+        })
```

Comment:
I propose to put the content of this if in a new function `identifyOnExternalWMSLayer`. What about WMTS not yet implemented, will be done later ?

Reply:
I have to check with the DI team, but AFAIK it doesn't make much sense to implement that for WMTS (it's always data that are richer, and tiled, such as background maps, cloud coverage etc...)

Reply:
I've created a ticket for the mixed projection, so that we can work on that later
https://jira.swisstopo.ch/browse/PB-363